File: deporvida_backend/src/main/java/com/deporvida/service/UsuarioService.java
package com.deporvida.service;

import com.deporvida.dto.request.ActualizarRolesRequest;
import com.deporvida.dto.request.DireccionRequest;
import com.deporvida.dto.response.DireccionResponse;
import com.deporvida.dto.response.PaginaResponse;
import com.deporvida.dto.response.UsuarioResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface UsuarioService {

    UsuarioResponse getById(Long id);

    UsuarioResponse getByNombreUsuario(String nombreUsuario);

    UsuarioResponse getByCorreo(String correo);

    PaginaResponse<UsuarioResponse> getAll(Pageable pageable);

    UsuarioResponse update(Long id, UsuarioResponse request);

    void delete(Long id);

    UsuarioResponse toggleActivo(Long id);

    List<DireccionResponse> getDirecciones(Long usuarioId);

    DireccionResponse addDireccion(Long usuarioId, DireccionRequest request);

    DireccionResponse updateDireccion(Long usuarioId, Long direccionId, DireccionRequest request);

    void deleteDireccion(Long usuarioId, Long direccionId);

    DireccionResponse setDireccionPredeterminada(Long usuarioId, Long direccionId);

    UsuarioResponse actualizarRoles(Long usuarioId, ActualizarRolesRequest request);
}

File: src/services/usuarioService.ts
import api from './api';

export interface UsuarioResponse {
  id: number;
  nombreUsuario: string;
  correo: string;
  nombres: string;
  apellidos: string;
  nombreCompleto: string;
  telefono: string;
  activo: boolean;
  roles: string[];
  fechaCreacion: string;
}

export interface ActualizarRolesRequest {
  roles: string[];
}

export interface PaginaResponse<T> {
  content: T[];
  totalPages: number;
  totalElements: number;
  size: number;
  number: number;
}

export const usuarioService = {
  getAll: (params?: { page?: number; size?: number }) =>
    api.get<PaginaResponse<UsuarioResponse>>('/usuarios', { params }),

  getById: (id: number) =>
    api.get<UsuarioResponse>(`/usuarios/${id}`),

  update: (id: number, data: Partial<UsuarioResponse>) =>
    api.put<UsuarioResponse>(`/usuarios/${id}`, data),

  delete: (id: number) =>
    api.delete(`/usuarios/${id}`),

  toggleActivo: (id: number) =>
    api.put<UsuarioResponse>(`/usuarios/${id}/toggle-activo`),

  actualizarRoles: (id: number, data: ActualizarRolesRequest) =>
    api.put<UsuarioResponse>(`/usuarios/${id}/roles`, data),
};
